### DISCUSSION TITLE
Bulk add with own tags for each item?

### DISCUSSION THREAD

**Discussion by dejvidq** [ArchiveBox/ArchiveBox#1628] [Q&A] [answered]
Hello, 
I have around 1100 links I'd like to add to my archivebox instance. I know that I can add those links to one file and then add it, but my problem is that each link might have tags, sometimes 0, but sometimes 7. Can I somehow perform similar bulk add, but with keeping those tags for those urls I already have tags for?

**Comment by pirate** [comment, marked as answer]
Several of the parsers support importing urls with tags & merging them with existing records, e.g.:

- https://github.com/ArchiveBox/ArchiveBox/blob/dev/archivebox/parsers/generic_rss.py (rss feed)
- https://github.com/ArchiveBox/ArchiveBox/blob/dev/archivebox/parsers/generic_json.py (big json list)
- https://github.com/ArchiveBox/ArchiveBox/blob/dev/archivebox/parsers/generic_jsonl.py (one json per line)
- https://github.com/ArchiveBox/ArchiveBox/blob/dev/archivebox/parsers/generic_html.py (nescape html bookmarks format)
- [more here...](https://github.com/ArchiveBox/ArchiveBox/tree/dev/archivebox/parsers)

If you have no particular preference, I recommend preparing a JSONL file like so:

`new_links.jsonl`:
```jsonl
{"url": "https://example.com/page1.html", "tags": "tag1,tag2,tag3", "created_at": 1735890129.25635}
{"url": "https://example.com/page2.html", "tags": "tag2,tag3", "created_at": 1735890130.12345}
...
```
```bash
archivebox add --parser=jsonl < new_links.jsonl
```

Whenever you import URLs, existing URLs that you have already archived previously wont be re-archived, but any new tags will be merged in and applied to the existing Snapshot (so it's safe to run an import on the same JSONL multiple times / every 24hr / etc.).

**Reply by dejvidq** [reply]
Great, thank you!
